src/librustc_resolve/lib.rs

```diff
@@ -2319,20 +2319,27 @@ impl<'a> Resolver<'a> {
                     } else if i == 0 {
                         (format!("use of undeclared type or module `{}`", ident), None)
                     } else {
-                        let mut msg =
-                            format!("could not find `{}` in `{}`", ident, path[i - 1].ident);
+                        // Do not emit `PathRoot`(`{{root}}`) as it's unclear.
+                        let ident_str;
+                        let ident_name = if path[i - 1].ident.name == kw::PathRoot {
+                            "the crate root"
```

Comment:
`kw::PathRoot` is a crate root only if its edition is 2015, otherwise the wording should be "could not find extern crate `{}`".

Reply:
The 2018 case also needs a test.
(That's unless we never get to here for 2018 edition `kw::PathRoot`, then it needs a comment.)